source/extensions/filters/http/cache/cache_header_utility.cc

```diff
@@ -0,0 +1,107 @@
+#include "extensions/filters/http/cache/cache_header_utility.h"
+
+#include <vector>
+
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/header_utility.h"
+#include "common/http/headers.h"
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+namespace {
+const absl::string_view bytes_{"bytes"};
+}
+
+std::vector<RawByteRange> CacheHeaderUtility::getRanges(const Http::HeaderMap& request_headers) {
+  if (request_headers.Method() == nullptr ||
```

Comment:
There's no need for getRange to check the method--calling it with headers that should have range headers is a programming error, and not getRange's problem.

Reply:
Changed to an ASSERT.